src/test/java/org/mule/test/extension/file/common/ConcurrentLockTestCase.java

```diff
@@ -148,4 +163,107 @@ protected PathLock createLock(Path path) {
     @Override
     public void changeToBaseDir() {}
   }
+
+  @Test
+  public void concurrentUriLock() throws Exception {
+    new Thread(() -> {
+      try {
+        mainThreadLatch.release();
+        secondaryThreadLatch.await(TIMEOUT, TIMEOUT_UNIT);
+        tryUriLock();
+      } catch (Exception e) {
+        throw new RuntimeException(e);
+      }
+    }).start();
+
+    mainThreadLatch.await(TIMEOUT, TIMEOUT_UNIT);
+    secondaryThreadLatch.release();
+    tryUriLock();
+
+    assertionLatch.await(TIMEOUT, TIMEOUT_UNIT);
+    assertThat(successful.get(), is(1));
+    assertThat(failed.get(), is(1));
+  }
+
+  private void tryUriLock() {
+    try {
+      if (externalFileSystem.lock(URI).tryLock()) {
+        successful.incrementAndGet();
+      } else {
+        failed.incrementAndGet();
+      }
+    } catch (Exception e) {
+      failed.incrementAndGet();
+    }
+    assertionLatch.countDown();
+  }
+
+  private class TestExternalFileSystem extends ExternalFileSystem {
```

Comment:
You could change TestFileSystem so that it is an ExternalFileSystem if you wanted to make the test class shorter.